drivers/I2C.cpp

```diff
@@ -125,6 +125,7 @@ void I2C::unlock() {
 int I2C::transfer(int address, const char *tx_buffer, int tx_length, char *rx_buffer, int rx_length, const event_callback_t& callback, int event, bool repeated)
 {
     lock();
+    sleep_manager_lock_deep_sleep();
     if (i2c_active(&_i2c)) {
```

Comment:
If I2C is already active, you aren't starting a new transfer.  You should probably either unlock before returning or add the lock until after the active check.